core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -272,13 +289,15 @@ private Object handleWriteOnOriginPrimary(InvocationContext ctx, DataWriteComman
       Address backup = getNextMember(cacheTopology);
       if (backup != null) {
          // error responses throw exceptions from JGroupsTransport
-         CompletableFuture<Map<Address, Response>> rpcFuture =
-            rpcManager.invokeRemotelyAsync(Collections.singletonList(backup), backupCommand, defaultSyncOptions);
+         List<Address> recipients = Collections.singletonList(backup);
+         assert recipients.size() == 1;
```

Comment:
I guess you can trust the `singletonList()` to have size equals to 1 all the time :)